x-pack/plugins/security_solution/cypress/test_plans/detection_response/prebuilt_rules/original_scenarios.md

```diff
@@ -0,0 +1,264 @@
+
+```Gherkin
+Given user has all available rules installed
+When user navigates to the Add Elastic Rules Page
+Then user should see a message indicating that all available rules have been installed
+And user should see a CTA that leads to the Rules Management Page
+And user should navigate back to Rules Management Page when clicking on the CTA
+```
+
+#### **Scenario: User is notified that all installed rules are up to date in the Rule Updates tab**
+
+```Gherkin
+Given user has all available rules installed
+When user navigates to the Rule Update
+Then user should see a message indicating that all installed rules are up to date
+```
+
+### Prebuilt Rules Installation
+
+#### **Scenario: User without any installed prebuilt rules can install `<amount>` prebuilt rules**
+
+```Gherkin
+Given a user that doesn’t have prebuilt rules installed
+When user navigates to Add Elastic Rules Page
+Then available prebuilt rules are displayed on Elastic Rules table
+And user can install <amount> prebuilt Rules
+And successfully installed message is displayed after installation
+And installed rules are removed from Elastic Rules table
+And rules to install counter is decreased accordingly
+
+Examples:
+  | amount     |
+  | all        |
+  | selected   |
+  | individual |
+```
+
+#### **Scenario: User navigating to the Add Elastic Rules page sees a loading skeleton until the prebuilt rules package installation is completed**
+
+```Gherkin
+Given a user that is on Rules Management Page
+When user to the Add Elastic Rules page before rules package is installed
+Then a loading skeleton is displayed until the installation is completed
+```
+
+### Prebuilt Rules Update
+
+#### **Scenario: Users can update prebuilt rules**
+
+```Gherkin
+Given user already has 8.x prebuilt rules installed in Kibana
+And there are new updates available for those prebuilt rules
+And user is on Rules Management Page
+When user navigates to the Rules Update tab
+Then user should see all the prebuilt rules that have updates available
+And user can update outdated prebuilt rules
+And successfully updated message is displayed
+And Rules Upgrade tab counter is decreased according to the number of updated rules
+```
+
+### Package Installation / Rule Installation / Rule Update Failure
+
+#### **Scenario: Error message is displayed when any prebuilt rules operation fails**
+
+```Gherkin
+Given user is <action> prebuilt rules
+When the installation or update process fails
+Then user should see an error message
+And prebuilt rules are not installed/updated
+And the callout message for pending installs/updates is still displayed on Rules Management Page
+And the number of available rules to install and upgrade in the badges does not change
+
+Examples:
+  | action                |
+  | installing all        |
+  | installing selected   |
+  | installing individual |
+  | updating all          |
+  | updating selected     |
+  | updating individual   |
+```
```

Comment:
Replaced by:

#### **Scenario: Error is handled when any operation on prebuilt rules fails**

**Coverage**: ?.

```Gherkin
Given user is <operation> prebuilt rules
When the operation fails
Then user should see an error message
And prebuilt rules should not be installed/upgraded
And the CTAs to install/upgrade prebuilt rules should remain on the Rule Management page
And the numbers of available rules to install/upgrade should not change

Examples:
  | operation             |
  | installing all        |
  | installing selected   |
  | installing individual |
  | upgrading all         |
  | upgrading selected    |
  | upgrading individual  |
```